### PR TITLE
add new agrument whether to remove core set from refined set

### PR DESCRIPTION
**Pull request by zhuhui-in**
add new argument "whether to remove core set from refined set", and test potentialnet performance on PDBBind core set when trained on refine/core set.

### CODE REVIEW

**Comment by zhuhui-in** [comment]
Training set: PDBBind v2015 refined set (removed core set) (3516 complexes)
Test set: PDBBind v2013 core set (195 complexes)
Splitting method: Random
| R^2 |self_test(random split)| paper(sequence split)|paper(structure split)|
|---|---|---|---|
|train| 0.823 |   |   |
|validation| 0.443  |   |   |
|test| 0.429  | 0.480  |  0.629 |


refined_core

**Comment by 0ut0fcontrol** [comment]
Is it ready for review?

**Comment by zhuhui-in** [comment]
> Is it ready for review?

yes！

File: python/dgllife/data/pdbbind.py [hunk 1]
```diff
@@ -78,19 +78,22 @@ class PDBBind(object):
...
-                 remove_hs=False, use_conformation=True,
+    def __init__(self, subset, pdb_version, load_binding_pocket=True, remove_coreset_from_refinedset=True, sanitize=False, 
+                 calc_charges=False, remove_hs=False, use_conformation=True,
                  construct_graph_and_featurize=ACNN_graph_construction_and_featurization,
-                 zero_padding=True, num_processes=None):
+                 zero_padding=True, num_processes=None, **kwargs):
         self.task_names = ['-logKd/Ki']
         self.n_tasks = len(self.task_names)
-        self._read_data_files(pdb_version, subset, load_binding_pocket)
+        self._read_data_files(pdb_version, subset, load_binding_pocket,remove_coreset_from_refinedset)
```
Comment (0ut0fcontrol) [inline python/dgllife/data/pdbbind.py]:
```suggestion
        self._read_data_files(pdb_version, subset, load_binding_pocket, remove_coreset_from_refinedset)
```

File: python/dgllife/data/pdbbind.py [hunk 2]
```diff
@@ -151,8 +154,23 @@ def _read_data_files(self, pdb_version, subset, load_binding_pocket):
...
+        
+        ##remove core set from refine set
         pdbs = self.df['PDB_code'].tolist()
+        if remove_coreset_from_refinedset:
+            if pdb_version == 'v2015':
+                with open(extracted_data_path + '/v2015/INDEX_core_data.2013','r') as f:
+                    for line in f:
+                        line = line.strip().split()
+                        if line[0] != "#" and line[0] in pdbs:
+                            pdbs.remove(line[0])
```
Comment (0ut0fcontrol) [inline python/dgllife/data/pdbbind.py]: use more accurate name.
```suggestion
                        fields = line.strip().split()
                        if line[0] != "#" and fields[0] in pdbs:
                            pdbs.remove(fields[0])
```

File: python/dgllife/data/pdbbind.py [hunk 3]
```diff
@@ -151,8 +154,23 @@ def _read_data_files(self, pdb_version, subset, load_binding_pocket):
...
+
+            if pdb_version == 'v2007':
+                with open (extracted_data_path + '/2007/INDEX.2017.core.data','r') as f:
+                    for line in f:
+                        line = line.strip().split()
```
Comment (0ut0fcontrol) [inline python/dgllife/data/pdbbind.py]: same in here.

**Review by 0ut0fcontrol — APPROVED** [review]
It seems the performance on the core set is right.
Waiting for a response from dgl-lifesci.

Meanwhile, you can start working on your own dataset.
